Assets/Scripts/Scheduler/EventPrototypes.cs

```diff
@@ -0,0 +1,43 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Xml;
+using Scheduler;
+
+public class EventPrototypes : XmlPrototypes<ScheduledEvent>
+{
+    public EventPrototypes() : base("Events.xml", "Events", "Event")
+    {
+    }
+
+    public void Add(string name, ScheduledEvent eventPrototype)
```

Comment:
```SetPrototype``` does something similar to this function. If you don't want it to replace an item that already exists, or give a warning, I would adding an ```AddPrototype``` function to ```BasePrototypes``` so that it can be used by any class, or change ```SetPrototype```.